lib/mayaUsd/ufe/UsdUndoRenameCommand.cpp

```diff
@@ -60,8 +60,23 @@ UsdUndoRenameCommand::UsdUndoRenameCommand(const UsdSceneItem::Ptr& srcItem, con
     }
     else
     {
-        auto layers = MayaUsdUtils::layersWithPrimSpec(_prim);
+        // account for internal vs external references
+        if (_prim.HasAuthoredReferences()) {
+            auto primSpec = MayaUsdUtils::getPrimSpecAtEditTarget(_stage, _prim);
+            for (const SdfReference& ref : primSpec->GetReferenceList().GetAddedOrExplicitItems()) {
+                // GetAssetPath returns the asset path to the root layer of the referenced layer
+                // this will be empty in the case of an internal reference.
+                if (ref.GetAssetPath().empty()) { 
+                    return;
+                }else{
+                   std::string err = TfStringPrintf("Unable to rename referenced object [%s]", 
+                                                    _prim.GetName().GetString().c_str());
+                   throw std::runtime_error(err.c_str());
+                }
+            }
+        }
 
```

Comment:
Using lower-level SdfReference to detect internal vs external reference to meet following condition:

`Internal references (references without a file path specified) from the same file, should be renamable and the references string updated. //Error: Unable to rename referenced object [object name]`

Reply:
Your first sentence (before //Error) would make a great code comment right after line 63.

Reply:
@ppt-adsk please see a305af95bbe17a8e7a68e8e807335ccc80fe043f